py_wrap_generator.py

```diff
@@ -0,0 +1,2101 @@
+		return ret
+
+	def gen_call_cpp(self):
+		ret = self.gen_varname()
+		if self.wtype.name.split(" ")[-1] in primitive_types or self.wtype.name in enum_names:
+			if self.wtype.attr_type == attr_types.star:
+				return "&" + ret
+			return ret
+		if self.wtype.name not in classnames:
+			if self.wtype.attr_type == attr_types.star:
+				return "&" + ret + "___tmp"
+			return ret + "___tmp"
+		if self.wtype.attr_type != attr_types.star:
+			ret = "*" + ret
+		return self.wtype.name + "::get_py_obj(" + self.gen_varname()  + ")"
+
+	#Generate cleanup code
+	def gen_cleanup(self):
+		if self.wtype.name in primitive_types or self.wtype.name in classnames or self.wtype.name in enum_names or not self.wtype.attr_type == attr_types.star or (self.wtype.name in known_containers and self.wtype.attr_type == attr_types.star):
+			return ""
+		return "\n\t\tdelete " + self.gen_varname() + "___tmp;"
+
+class WClass:
+	name = None
+	namespace = None
+	link_type = None
+	id_ = None
+	string_id = None
+	hash_id = None
+	needs_clone = False
+	found_funs = []
+	found_vars = []
+	found_constrs = []
+
+	def __init__(self, name, link_type, id_, string_id = None, hash_id = None, needs_clone = False):
+		self.name = name
+		self.namespace = None
+		self.link_type = link_type
+		self.id_ = id_
+		self.string_id = string_id
+		self.hash_id = hash_id
+		self.needs_clone = needs_clone
+		self.found_funs = []
+		self.found_vars = []
+		self.found_constrs = []
+
+	def printable_constrs(self):
+		ret = 0
+		for con in self.found_constrs:
+			if not con.protected:
+				ret += 1
+		return ret
+
+	def gen_decl(self, filename):
+		long_name = self.namespace + "::" + self.name
+
+		text = "\n\t// WRAPPED from " + filename
+		text += "\n\tstruct " + self.name
+		if self.link_type == link_types.derive:
+			text += " : public " + self.namespace + "::" + self.name
+		text += "\n\t{\n"
+
+		if self.link_type != link_types.derive:
+
+			text += "\t\t" + long_name + "* ref_obj;\n"
+
+			if self.link_type == link_types.ref_copy or self.link_type == link_types.pointer:
+				text += "\n\t\t" + long_name + "* get_cpp_obj() const\n\t\t{\n\t\t\treturn ref_obj;\n\t\t}\n"
+			elif self.link_type == link_types.global_list:
+				text += "\t\t" + self.id_.wtype.name + " " + self.id_.varname + ";\n"
+				text += "\n\t\t" + long_name + "* get_cpp_obj() const\n\t\t{"
+				text += "\n\t\t\t" + long_name + "* ret = " + long_name + "::get_all_" + self.name.lower() + "s()->at(this->" + self.id_.varname + ");"
+				text += "\n\t\t\tif(ret != NULL && ret == this->ref_obj)"
+				text += "\n\t\t\t\treturn ret;"
+				text += "\n\t\t\tthrow std::runtime_error(\"" + self.name + "'s c++ object does not exist anymore.\");"
+				text += "\n\t\t\treturn NULL;"
+				text += "\n\t\t}\n"
+
+			#if self.link_type != link_types.pointer:
+			text += "\n\t\tstatic " + self.name + "* get_py_obj(" + long_name + "* ref)\n\t\t{"
+			text += "\n\t\t\t" + self.name + "* ret = (" + self.name + "*)malloc(sizeof(" + self.name + "));"
```

Comment:
Could new be used instead of malloc & cast?

Reply:
Actually no, since new would call the constructor and for classes that do not contain a default constructor this will not work.